src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> converteData(List<String> dataFromFile) {
+        List<FruitTransaction> convertedData = new ArrayList<>();
+        for (String data : dataFromFile) {
+            if (data != null && !data.trim().isEmpty()) {
+                String[] eachLineElement = data.split(",");
+                if (eachLineElement.length == 3) {
+                    convertedData.add(new FruitTransaction(eachLineElement[0], eachLineElement[1],
```

Comment:
Be cautious with the `Integer.parseInt` method, as it can throw a `NumberFormatException` if the input is not a valid integer. Consider adding a try-catch block to handle this potential exception and provide a more informative error message.